app/models/openstax/accounts/account.rb

```diff
@@ -46,12 +49,36 @@ def has_authenticated?
       !access_token.nil?
     end
 
+    def valid_openstax_uid?
+      !openstax_uid.nil? && openstax_uid > 0
+    end
+
+    def openstax_uid=(val)
+      super
+
+      set_account_type(force: true)
+    end
+
+    def set_faculty_status
+      self.faculty_status ||= :no_faculty_info
+    end
+
+    def set_account_type(force: false)
+      return account_type if account_type.present? && !force
+
+      self.account_type = valid_openstax_uid? ? :remote : :local
+    end
```

Comment:
do we need an account_type enum if it is always just based on validity of openstax_uid?